samples/gatsby-preview/src/index.jsx

```diff
@@ -0,0 +1,157 @@
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+
+    const parameters = this.props.sdk.parameters
+    const projectId = parameters.installation.projectId
+    
+    const { contentTypeSlug, isAutoUpdate } = parameters.instance
+
+    this.state = {
+      projectId,
+      projectUrl:
+        "https://" + projectId + "-preview.gtsb.io/" + contentTypeSlug + "/",
+      isAutoUpdate: isAutoUpdate,
+      webhookUrl:
+        "https://backends.ctffns.net/gatsby-preview-proxy/" + projectId,
+      debounceInterval: null,
+      isUpdated: false
+    }
+  }
+
+  componentDidMount = () => {
+    
+    this.detachFn = this.props.sdk.entry.onSysChanged(this.onSysChanged)
+
+    this.props.sdk.window.startAutoResizer()
+
+    this.state.debounceInterval = setInterval(this.refreshGatsbyPreview, 1000)
+  }
+
+  componentWillUnmount = () => {
+    this.detachFn()
+    clearInterval(this.state.debounceInterval)
+  }
+
+  onError = error => {
+    this.setState({ working: false })
+    this.props.sdk.notifier.error(error.message)
+  }
+
+  onSysChanged = () => {
+    if (!this.state.isAutoUpdate) {
+      return
+    }
+    this.state.isUpdated = true
+  }
+
+  refreshGatsbyPreview = () => {
+    if (!this.state.isUpdated) {
```

Comment:
You should also check `isAutoUpdate`

Reply:
this method is also invoked with a button, that is why I am not checking